third_party/cyw30739_sdk/compress_cgs_data.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+
+import argparse
+import io
+import pathlib
+import re
+import subprocess
+import sys
+
+from collections import namedtuple
+
+
+Item = namedtuple('Item', 'line type nested_data')
+Entry = namedtuple('Entry', 'items name remark')
+CommentedBytes = namedtuple('CommentedBytes', 'items data')
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--output", required=True, type=pathlib.Path)
+    parser.add_argument("--input", required=True, type=pathlib.Path)
+    parser.add_argument("--lzss_tool", required=True, type=pathlib.Path)
+
+    option = parser.parse_args()
+
+    cgs = parse_cgs(option.input)
+    compress_cgs(option, cgs)
+    export_cgs(option.output, cgs)
+
+
+def parse_cgs(file_name):
+    cgs = []
+
+    with open(file_name, mode="r") as file:
+        while True:
+            line = file.readline()
+            if len(line) == 0:
+                break
+
+            striped_line = line.strip()
+            item_type = None
+            nested_data = None
+
+            if striped_line.startswith('ENTRY'):
+                item_type = "entry"
+                nested_data = parse_entry(file, line)
+            elif striped_line.startswith('#') or striped_line.startswith('//'):
+                item_type = "comment"
+            elif striped_line.startswith('DEFINITION'):
+                item_type = "definition"
+
+            cgs.append(Item(line, item_type, nested_data))
+
+    return cgs
+
+
+def parse_entry(file: io.TextIOBase, line):
+
+    data_re = re.compile(r'^\s*ENTRY\s+"([^"]+)"(?:\s*=\s*"([^"]+)")?$')
```

Comment:
you may want to consider writing some parser like lark based parsers (IDL parsers use this). this is reasonably hard to follow: at least a comment explaining what this parses and how would be helpful: example of how a line looks like and what parsing does.